scripts/demangle.py

```diff
@@ -1,104 +1,170 @@
-        print(f"ERROR: Malformed function definition for '{func_name}' in code file")
-        return f"ERROR: Malformed function definition for '{func_name}' in code file"
 
-    function_code = code_content[func_start:func_now]
-    return func_init+function_code
+# get source code from docker
+def get_source_from_docker(
+    file_path: str, file_name: str, proj_name: str
+) -> Optional[str]:
+    # Create a container from the image
+    client = docker.from_env()
+    image = client.images.get(f"gcr.io/oss-fuzz/{proj_name}:latest")
+    container = client.containers.run(image, detach=True)
+    # check file_path, it may not the absolut path
+    if "/" not in file_path:
+        file_path = os.path.join("/src", proj_name, file_path)
+    try:
+        code_content = ""
+        for chunk in container.get_archive(file_path)[0]:
+            code_content += chunk.decode("utf-8")
+    except docker.errors.NotFound:
+        code_content = None
+        error(f" File  {file_path}  NOT found\n")
+    # Stop and remove the container
+    container.stop()
+    container.remove()
+    return code_content
 
-def demangle(func: str):
-    # Load the C++ Standard Library
-    libcxx = ctypes.cdll.LoadLibrary("libc++.so.1")
 
-    # Define the signature of the __cxa_demangle() function
-    demangle = libcxx.__cxa_demangle
-    demangle.argtypes = [ctypes.c_char_p, ctypes.c_char_p, ctypes.POINTER(ctypes.c_size_t), ctypes.POINTER(ctypes.c_int)]
-    # Call __cxa_demangle() to demangle the input string
-    output_buffer = ctypes.create_string_buffer(256)
-    output_size = ctypes.c_size_t(256)
-    status = ctypes.c_int(0)
-    result = demangle(func.encode(), output_buffer, ctypes.byref(output_size), ctypes.byref(status))
-    # Forced return demangle value
-    return output_buffer.value.decode('utf-8')
-    # if result == 0:
-    #     # Demangling was successful
-    #     demangled_name = output_buffer.value.decode('utf-8')
-    #     return demangled_name
-    # else:
-    #     # Demangling failed
-    #     print("Demangling failed with status", status)
+def main(proj_name: str):
+    file_func_delim = "?"
+    json_path = os.path.join(OSSFUZZ_SCRIPTS_HOME, "dump", proj_name)
+    json_file_names = [f for f in os.listdir(json_path) if f.endswith(".json")]
+    info(f"Looking for json file: {json_file_names}")
 
-for json_file_name in file_names:
-    file_name = json_file_name.split('.json')[0]
-    # Find file path
-    if len(glob.glob(proj_path+'/'+file_name+'.c*'))==0:
-        print('ERROR: File '+proj_path+'/'+file_name+'.c*'+" NOT found")
-        continue
-    file_path = glob.glob(proj_path+'/'+file_name+'.c*')[0]
-    # if file_name == 'augeas_api_fuzzer': continue
-    # open Json file and get filename + func name
-    with open(json_path+'/'+json_file_name, 'r') as f:
-        try:
-            data = json.load(f)
-        except:
+    for json_file_name in json_file_names:
+        file_name = json_file_name.split(".json")[0]
```

Comment:
Use `os.path.splitext` to prevent a file with multiple `.json` extensions.
https://stackoverflow.com/questions/678236/how-do-i-get-the-filename-without-the-extension-from-a-path-in-python